Tests/MongoSwiftSyncTests/SpecTestRunner/TestOperation.swift

```diff
@@ -335,74 +525,101 @@ struct UpdateMany: TestOperation {
+    let session: String
     let filter: Document
     let options: DeleteOptions
 
-    private enum CodingKeys: String, CodingKey { case filter }
+    private enum CodingKeys: String, CodingKey { case session, filter }
 
     init(from decoder: Decoder) throws {
         self.options = try DeleteOptions(from: decoder)
         let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.session = (try? container.decode(String.self, forKey: .session)) ?? ""
         self.filter = try container.decode(Document.self, forKey: .filter)
     }
 
-    func execute(on target: TestOperationTarget, session: ClientSession?) throws -> TestOperationResult? {
+    func execute(on target: TestOperationTarget, sessionDict: [String: ClientSession]) throws -> TestOperationResult? {
         guard case let .collection(collection) = target else {
             throw TestError(message: "collection not provided to deleteMany")
         }
-        let result = try collection.deleteMany(self.filter, options: self.options, session: session)
+        let result = try collection.deleteMany(self.filter, options: self.options, session: sessionDict[self.session])
         return TestOperationResult(from: result)
     }
 }
 
 struct DeleteOne: TestOperation {
+    let session: String
     let filter: Document
     let options: DeleteOptions
 
-    private enum CodingKeys: String, CodingKey { case filter }
+    private enum CodingKeys: String, CodingKey { case session, filter }
 
     init(from decoder: Decoder) throws {
         self.options = try DeleteOptions(from: decoder)
         let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.session = (try? container.decode(String.self, forKey: .session)) ?? ""
         self.filter = try container.decode(Document.self, forKey: .filter)
     }
 
-    func execute(on target: TestOperationTarget, session: ClientSession?) throws -> TestOperationResult? {
+    func execute(on target: TestOperationTarget, sessionDict: [String: ClientSession]) throws -> TestOperationResult? {
         guard case let .collection(collection) = target else {
             throw TestError(message: "collection not provided to deleteOne")
         }
-        let result = try collection.deleteOne(self.filter, options: self.options, session: session)
+        let result = try collection.deleteOne(self.filter, options: self.options, session: sessionDict[self.session])
         return TestOperationResult(from: result)
     }
 }
 
 struct InsertOne: TestOperation {
+    let session: String
     let document: Document
 
-    func execute(on target: TestOperationTarget, session _: ClientSession?) throws -> TestOperationResult? {
+    private enum CodingKeys: String, CodingKey { case session, document }
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.session = (try? container.decode(String.self, forKey: .session)) ?? ""
```

Comment:
rather than nil coalescing, I think its more idiomatic to just store a `String?` on each of these operations. Also, instead of doing `try?`, we can use [`decodeIfPresent`](https://developer.apple.com/documentation/swift/keyeddecodingcontainer/2893445-decodeifpresent) to conditionally decode the session name.

Reply:
Done.